### DISCUSSION TITLE
How to pull stackable spark docker image locally?

### DISCUSSION THREAD

**Discussion by paulpaul1076** [stackabletech/spark-k8s-operator#314] [Q&A] [answered]
I have a bunch of dependency problems when running spark and including the spark-sql-kafka jar. I wanted to debug what the problem is, and for that I wanted to download the jars from stackable spark docker image from here:

https://repo.stackable.tech/#browse/browse:docker:v2%2Fstackable%2Fspark-k8s%2Ftags%2F3.3.0-stackable23.7.0

But when I do that I get an error:

```
Error response from daemon: pull access denied for stackable/spark-k8s, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
```

How do I get it to pull?

**Comment by sbernauer** [comment, marked as answer]
Hi @paulpaul1076!
The error message seems to indicate that you are trying to pull from (the official) docker hub https://hub.docker.com/:
```
docker pull stackable/spark-k8s:3.3.0-stackable23.7.0          
Error response from daemon: pull access denied for stackable/spark-k8s, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
```

We self-host our own container registry, the image is called e.g. `docker.stackable.tech/stackable/spark-k8s:3.3.0-stackable23.7.0`.

You should be able to pull that image as follows
```
docker pull docker.stackable.tech/stackable/spark-k8s:3.3.0-stackable23.7.0 
3.3.0-stackable23.7.0: Pulling from stackable/spark-k8s
28ff5ee6facb: Downloading [=>                                                 ]  1.223MB/39.35MB
```
